posthog/session_recordings/session_recording_helpers.py

```diff
@@ -135,6 +139,7 @@ def preprocess_replay_events(
     session_id = events[0]["properties"]["$session_id"]
     window_id = events[0]["properties"].get("$window_id")
     snapshot_source = events[0]["properties"].get("$snapshot_source", "web")
+    snapshot_library = events[0]["properties"].get("$snapshot_library", user_agent)
```

Comment:
from a clickhouse review perspective - we'll take the provided snapshot library or the user agent from the request. the mobile libraries send custom user agents so we can probe for library in the long tail of SDK versions

also means this definitely isn't a low cardinality string

Reply:
instead of using the whole `user_agent` here making it somewhat high cardinality, can we actually parse the user agent and find out the SDK and default to web if not?

Reply:
yep. we can check if the user agent contains posthog, since the web sdk doesn't change the user agent